### PR TITLE
Avoid using a `Task` when code is pure. Prefer to use `Try`

### PR DESCRIPTION
**Pull request by guizmaii**

### CODE REVIEW

File: zio-kafka/src/main/scala/zio/kafka/admin/AdminClient.scala [hunk 1]
```diff
@@ -1470,15 +1494,32 @@ object AdminClient {
...
     def toScala: Option[T] = if (v.isPresent) Some(v.get()) else None
   }
 
-  implicit final class OptionOps[T](val v: Option[T]) extends AnyVal {
+  implicit final class OptionOps[T](private val v: Option[T]) extends AnyVal {
     def toJava: Optional[T] = v.fold(Optional.empty[T])(Optional.of)
   }
+
+  implicit final class ListOps[A](private val list: List[A]) extends AnyVal {
+    def forEach[B](f: A => Try[B]): Try[List[B]] = {
```
Comment (svroonland) [inline zio-kafka/src/main/scala/zio/kafka/admin/AdminClient.scala:1510]: It's more scala/cats idomatic to have a `List[Try[A]]` and a `sequence: Try[List[A]]` method. Have a look at `Future.sequence` for example.

File: zio-kafka/src/main/scala/zio/kafka/admin/AdminClient.scala [hunk 2]
```diff
@@ -1470,15 +1494,32 @@ object AdminClient {
       ZIO.succeed(client.close(settings.closeTimeout))
     )
 
-  implicit final class MapOps[K1, V1](val v: Map[K1, V1]) extends AnyVal {
+  implicit final class MapOps[K1, V1](private val v: Map[K1, V1]) extends AnyVal {
```
Comment (svroonland) [inline zio-kafka/src/main/scala/zio/kafka/admin/AdminClient.scala:1497]: IMO the `private` has no added value for these implicit classes, whose only purpose is to define extension methods.

**Review by svroonland — CHANGES_REQUESTED** [review]
Looks fine overall, but a few points

File: zio-kafka/src/main/scala/zio/kafka/admin/AdminClient.scala (hunk 1 above)
Comment (guizmaii) [reply zio-kafka/src/main/scala/zio/kafka/admin/AdminClient.scala:1510]: `forEach` comes from zio-prelude ;)

File: zio-kafka/src/main/scala/zio/kafka/admin/AdminClient.scala (hunk 2 above)
Comment (guizmaii) [reply zio-kafka/src/main/scala/zio/kafka/admin/AdminClient.scala:1497]: This is just copying how cats and co. are defining implicit class. I know they did a lot of research to find the best way to define them to be sure there is not overhead at runtime, I'm just copying their way of coding here

Comment (svroonland) [reply zio-kafka/src/main/scala/zio/kafka/admin/AdminClient.scala:1497]: Ah I was not aware of that. Alright

File: zio-kafka/src/main/scala/zio/kafka/admin/AdminClient.scala (hunk 1 above)
Comment (svroonland) [reply zio-kafka/src/main/scala/zio/kafka/admin/AdminClient.scala:1510]: Right, makes sense.

**Review by svroonland — APPROVED** [review]

**Review by jdegoes — APPROVED** [review]

**Pull request merged**